packages/noya-state/src/reducers/pointReducer.ts

```diff
@@ -97,203 +91,46 @@ export function pointReducer(
-          const boundingRect = boundingRects[layer.do_objectID];
-
-          if (!Layers.isPointsLayer(layer) || !boundingRect) return;
-
-          const curveMode = layer.points[pointIndex].curveMode;
-
-          // Update all points by first transforming to the canvas's coordinate system
-          const decodedPoints = layer.points.map((curvePoint) =>
-            decodeCurvePoint(curvePoint, boundingRect),
-          );
-
-          const decodedPoint = decodedPoints[pointIndex];
-
-          const oppositeControlPointType =
-            controlPointType === 'curveFrom' ? 'curveTo' : 'curveFrom';
-
-          const controlPoint = decodedPoint[controlPointType];
-          const oppositeControlPoint = decodedPoint[oppositeControlPointType];
-
-          const selectedControlPointValue =
-            mode === 'replace' ? amount : controlPoint[axis] + amount;
-
-          const delta = controlPoint[axis] - selectedControlPointValue;
-
-          const oppositeControlPointDistance = distance(
-            decodedPoint.point,
-            oppositeControlPoint,
-          );
-
-          switch (curveMode) {
-            case Sketch.CurveMode.Mirrored:
-              controlPoint[axis] = selectedControlPointValue;
-
-              oppositeControlPoint[axis] += delta;
-              break;
-            case Sketch.CurveMode.Asymmetric:
-              controlPoint[axis] = selectedControlPointValue;
-
-              let theta =
-                Math.atan2(
-                  controlPoint.y - decodedPoint.point.y,
-                  controlPoint.x - decodedPoint.point.x,
-                ) + Math.PI;
-
-              const oppositeControlPointValue = {
-                x:
-                  oppositeControlPointDistance * Math.cos(theta) +
-                  decodedPoint.point.x,
-                y:
-                  oppositeControlPointDistance * Math.sin(theta) +
-                  decodedPoint.point.y,
-              };
-
-              decodedPoint[
-                oppositeControlPointType
-              ] = oppositeControlPointValue;
-              break;
-            default:
-              controlPoint[axis] = selectedControlPointValue;
-          }
-
-          const [minX, minY, maxX, maxY] = path(
-            CanvasKit,
-            layer.points,
-            layer.frame,
-          ).computeTightBounds();
-
-          const newRect: Rect = {
-            x: minX,
-            y: minY,
-            width: maxX - minX,
-            height: maxY - minY,
-          };
-
-          layer.frame = {
-            ...layer.frame,
-            ...newRect,
-          };
-
-          // Transform back to the range [0, 1], using the new bounds
-          const encodedPoints = decodedPoints.map((decodedCurvePoint) =>
-            encodeCurvePoint(decodedCurvePoint, layer.frame),
-          );
-
-          layer.points = encodedPoints;
-        });
+        if (!draft.selectedControlPoint) return state;
```

Comment:
We don't need to return `state` within an immer `produce`. It can just be `return`.

We can also remove `if (!state.selectedControlPoint) return state;` above if we check down here